src/rust/en.tsumino/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+#![no_std]
+
+use aidoku::{
+	error::Result,
+	helpers::uri::encode_uri,
+	prelude::*,
+	std::net::HttpMethod,
+	std::net::Request,
+	std::{String, Vec},
+	Chapter, DeepLink, Filter, FilterType, Listing, Manga, MangaPageResult, Page,
+};
+extern crate alloc;
+use alloc::{string::ToString, vec};
+mod helper;
+mod parser;
+
+#[get_manga_list]
+fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
+	let mut query: String = String::new();
+	let mut sort: String = String::from("Newest");
+	let mut tags: String = String::new();
+	let mut i = 0;
+	for filter in filters {
+		match filter.kind {
+			FilterType::Title => {
+				query.push_str("&Text=");
+				query.push_str(&encode_uri(filter.value.as_string()?.read()));
+			}
+			FilterType::Genre => {
+				let tpe = 1;
+				tags.push_str(&format!(
+					"&Tags[{}][Type]={}",
+					i.to_string(),
+					tpe.to_string()
+				));
+				tags.push_str(&format!("&Tags[{}][Text]={}", i.to_string(), filter.name));
+				match filter.value.as_int().unwrap_or(-1) {
+					0 => tags.push_str(&format!("&Tags[{}][Exclude]=true", i.to_string())),
+					1 => tags.push_str(&format!("&Tags[{}][Exclude]=false", i.to_string())),
+					_ => continue,
+				}
+				i += 1;
+			}
+			// might remove this later in favor of the page listing
+			FilterType::Sort => {
+				let value = match filter.value.as_object() {
+					Ok(value) => value,
+					Err(_) => continue,
+				};
+				let index = value.get("index").as_int().unwrap_or(0) as i32;
+				let option = match index {
+					0 => "Newest",
+					1 => "Oldest",
+					2 => "Alphabetical",
+					3 => "Rating",
+					4 => "Pages",
+					5 => "Views",
+					6 => "Random",
+					7 => "Comments",
+					8 => "Popularity",
+					_ => continue,
+				};
+				sort = String::from(option)
+			}
+			_ => continue,
+		}
+	}
+	let url = String::from("https://www.tsumino.com/search/operate/");
+	let mut parameters = String::new();
+	parameters.push_str("PageNumber=");
+	parameters.push_str(&encode_uri(page.to_string()));
+	parameters.push_str(&query);
+	parameters.push_str("&Sort=");
+	parameters.push_str(&encode_uri(sort));
+	parameters.push_str(&tags);
+	let request = Request::new(&url, HttpMethod::Post)
+		.header("User-Agent", "Aidoku")
+		.body(parameters);
+	let json = request.json()?.as_object()?;
+	let data = json.get("data").as_array()?;
+
+	let mut manga_arr: Vec<Manga> = Vec::new();
+	for manga in data {
+		let obj = manga.as_object()?;
```

Comment:
probably best not to unwrap with `?` in cases like this since the whole function will fail if one manga is formatted wrong. if you're *positive* this won't fail, use `.expect()`, but otherwise just match it and `continue` if it's not an object.